src/c2goto/library/semaphore_lib.c

```diff
@@ -0,0 +1,106 @@
+#include <semaphore.h>
+
+#define __ESBMC_sem_lock_field(a) ((a).__lock)
+#define __ESBMC_sem_count_field(a) ((a).__count)
+#define __ESBMC_sem_init_field(a) ((a).__init)
+
+/************************* Sem manipulation routines ************************/
+
+int sem_init(sem_t *__sem, int __pshared, unsigned int __value)
+{
+  __ESBMC_atomic_begin();
+  __ESBMC_assert(!__pshared, "ESBMC does not check multiprocesses for now");
```

Comment:
I think this assertion might be misleading to user code working with semaphores, but I'm not 100% sure. The man-page says
>  A process-shared  semaphore  must  be  placed  in a shared memory region (e.g., a System V
              shared memory segment created using shmget(2), or a POSIX  shared  memory  object  built
              created using shm_open(3)).

As we don't support shared memory, yet, it's impossible to check this condition. But we also don't check certain validity conditions for other OMs (e.g., `fscanf` ignores its first argument). I would suggest to ignore the value of `pshared` until `shmget()` and friends are modelled. Maybe add a comment about this instead?